File: 792-binary-search/binary-search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        #implementing binary search in python 
        n = len(nums)
        s=0 
        e = n-1 
        while (s<=e):
            mid = (s+ (e-s)//2)
            value = nums[mid]
        
            if (value == target):
                return mid
            elif (value>target):
                e = mid -1 
            else : s  = mid + 1
        
        return -1 
        

File: 792-binary-search/binary-search.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var search = function(nums, target) {
    let n = nums.length  ; 
    var ans = -1 ; 
    let start =0, end =n-1 ; 
    while (start<=end ){
        let mid = (Math.floor(start+ (end-start)/2)); 
        let value = nums[mid];
        if (value == target ){
            ans = mid ; 
            break ;
            
        }
        else if (value > target){
            end = mid-1 ; 

        }else{
            start = mid + 1;
        }
    }

    return ans;
    
};